modules/shapefile/src/lib/parsers/parse-shapefile.ts

```diff
@@ -11,39 +22,34 @@ import {zipBatchIterators} from '../streaming/zip-batch-iterators';
 import {SHPLoader} from '../../shp-loader';
 import {DBFLoader} from '../../dbf-loader';
 
-type Feature = any;
-interface ShapefileOutput {
-  encoding?: string;
-  prj?: string;
-  shx?: SHXOutput;
-  header: SHPHeader;
-  data: object[];
-}
 /**
  * Parsing of file in batches
  */
 // eslint-disable-next-line max-statements, complexity
 export async function* parseShapefileInBatches(
   asyncIterator: AsyncIterable<ArrayBuffer> | Iterable<ArrayBuffer>,
-  options?: ShapefileLoaderOptions,
-  context?: LoaderContext
-): AsyncIterable<ShapefileOutput> {
+  options: ShapefileLoaderOptions,
+  context: LoaderContext
+): AsyncIterable<ShapefileBatchOutput> {
   const {reproject = false, _targetCrs = 'WGS84'} = options?.gis || {};
   const {shx, cpg, prj} = await loadShapefileSidecarFiles(options, context);
+  // We don't use shape yet
+  // const shape = options?.gis?.format || options?.shapefile?.shape || 'geojson';
 
   // parse geometries
   // @ts-ignore context must be defined
-  const shapeIterable: any = await context.parseInBatches(asyncIterator, SHPLoader, options);
+  const shapeIterable: AsyncIterator<(BinaryGeometry | null)[] | SHPHeader> =
+    await context.parseInBatches(asyncIterator, SHPLoader, options);
 
   // parse properties
-  let propertyIterable: any;
+  let propertyIterable: AsyncIterator<ObjectRowTableBatch> | undefined;
   // @ts-ignore context must be defined
   const dbfResponse = await context.fetch(replaceExtension(context?.url || '', 'dbf'));
   if (dbfResponse.ok) {
     // @ts-ignore context must be defined
     propertyIterable = await context.parseInBatches(dbfResponse, DBFLoader, {
       ...options,
-      dbf: {encoding: cpg || 'latin1'}
+      dbf: {encoding: cpg || options?.dbf?.encoding || 'latin1', shape: 'object-row-table'}
```

Comment:
Nit: This seems a bit messy, but maybe OK for now...

Reply:
Previously we defined `options.dbf.encoding` for the DBFLoader but ignored it in the ShapefileLoader. So if the CPG file didn't exist it would always use `latin1`. I should add this to the ShapefileLoader docs page too. https://github.com/visgl/loaders.gl/blob/master/modules/shapefile/docs/api-reference/dbf-loader.md#options